### DISCUSSION TITLE
How to compare CUTLASS with CUBLAS

### DISCUSSION THREAD

**Discussion by puddingfjz** [NVIDIA/cutlass#367] [Q&A] [answered]
Hi, 

I am new to both CUTLASS and CUBLAS. I want to know is there any method provided by cutlass that I can directly compare the performance of cublas and cutlass?

Thanks a lot!

**Comment by hwu36** [comment, marked as answer]
Sorry for the late response.  I am just back from my vacation.

You can use cutlass profiler to measure cutlass gemm performance: https://github.com/NVIDIA/cutlass/blob/master/media/docs/profiler.md

As to cublas, you can check this https://docs.nvidia.com/cuda/cublas/index.html#example-code

We just updated the performance comparison against cublas today: https://github.com/NVIDIA/cutlass#performance.  CUTLASS is awesome.

**Comment by masahi** [comment]
Actually, the easiest way is to use TVM, since it supports offloading to both cutlass and cublas. 

See https://github.com/apache/tvm/blob/dc988b288d85660822e0fbadbf1fc74326e763e5/tests/python/contrib/test_cutlass.py#L262-L275

**Comment by puddingfjz** [comment]
Thanks for all the advice!
